### PR TITLE
RESTEASY-2494 update resteasy-undertow-spring integration section

### PR DESCRIPTION
**Pull request by liweinan**

### CODE REVIEW

**Comment by liweinan** [comment]
In this PR:

- RESTEasy Spring Undertow Integration example is updated into the document.
- Section orders in Spring Document Chapter are adjusted to make it more readable.
- Multiple typos and missing contents are fixed/added.

File: docbook/reference/en/en-US/modules/RESTEasy_Spring_Integration.xml [hunk 1]
```diff
@@ -425,91 +376,138 @@ server.deploy(deployment);]]></programlisting>
...
+            supported as
+            is the use of <code>javax.servlet.http.HttpServletRequest</code> and
+            <code>javax.servlet.http.HttpServletResponse</code>
+            as method parameters.
+        </para>
+    </section>
+    <section id="SpringBootStarter">
+        <title>Spring Boot starter</title>
+        <para>
+            The RESTEasy project does not include its own component for Spring Boot integration, however PayPal has
```
Comment (asoldano) [inline docbook/reference/en/en-US/modules/RESTEasy_Spring_Integration.xml]: I'm thinking that while we're here we should revisit this section of the doc, given the Spring Boot Integration is now at https://github.com/resteasy/resteasy-spring-boot . Can you do that Weinan please?

Comment (liweinan) [reply docbook/reference/en/en-US/modules/RESTEasy_Spring_Integration.xml]: okay!

**Comment by liweinan** [comment]
@asoldano Because this PR is focusing on the topic to update `resteasy-undertow-spring` section so I created a separate issue to track the task for updating the doc for `spring-boot-starter` section: https://issues.redhat.com/browse/RESTEASY-2502

wdyt? If it's okay I'll keep this PR as-is and then start on the new issue.

**Comment by asoldano** [comment]
> @asoldano Because this PR is focusing on the topic to update `resteasy-undertow-spring` section so I created a separate issue to track the task for updating the doc for `spring-boot-starter` section: https://issues.redhat.com/browse/RESTEASY-2502
> 
> wdyt? If it's okay I'll keep this PR as-is and then start on the new issue.

OK, I'm fine with the proposed approach.

**Pull request merged**